examples/discovery-search-app/cypress/integration/pagination/spec.ts

```diff
@@ -0,0 +1,212 @@
+        cy.get('.bx--pagination__button--forward').click();
+      });
+
+      it('correctly requests the next page', () => {
+        cy.wait('@multiPageQuery')
+          .its('requestBody.offset')
+          .should('eq', 5);
+      });
+
+      describe('and the previous page arrow is clicked', () => {
+        beforeEach(() => {
+          cy.wait('@multiPageQuery');
+          cy.get('.bx--pagination__button--backward').click();
+        });
+
+        it('correctly requests the previous page', () => {
+          cy.wait('@multiPageQuery')
+            .its('requestBody.offset')
+            .should('eq', 0);
+        });
+      });
+    });
+
+    describe('and we use the page selector', () => {
+      beforeEach(() => {
+        cy.wait('@multiPageQuery');
+      });
+
+      it('we can navigate to each page of results', () => {
+        for (let i = 0; i < 12; i++) {
+          cy.get('.bx--pagination__right')
+            .find('.bx--select-input')
+            .select(`${i + 1}`);
+          cy.wait('@multiPageQuery')
+            .its('requestBody.offset')
+            .should('eq', i * 5);
+        }
+      });
+    });
+
+    describe('and we navigate to the last page of results', () => {
+      beforeEach(() => {
+        cy.wait('@multiPageQuery');
+        cy.get('.bx--pagination__right')
+          .find('.bx--select-input')
+          .select('12');
+      });
+
+      it('the next page button is disabled', () => {
+        cy.wait('@multiPageQuery');
+        cy.get('.bx--pagination__button--forward').should('be.disabled');
+      });
+
+      describe('and we increase the number of results per page to 50', () => {
+        beforeEach(() => {
+          cy.get('.bx--pagination__left')
+            .find('.bx--select-input')
+            .select('50');
+        });
+
+        it('returns to the first page, with the correct size', () => {
+          cy.wait('@multiPageQuery');
+          cy.wait('@multiPageQuery').then(xhr => {
+            expect(xhr.requestBody.count).to.eq(50);
+            expect(xhr.requestBody.offset).to.eq(0);
+          });
+          cy.get('.bx--pagination__right')
+            .find('.bx--select-input')
+            .should('have.value', '1');
+        });
+      });
+    });
+
+    describe('and items per page is set to 5', () => {
+      beforeEach(() => {
+        cy.get('.bx--pagination__left')
+          .find('.bx--select-input')
+          .select('5');
+      });
+
+      it('makes a request for 5 results', () => {
+        cy.wait('@multiPageQuery'); // not the request we're looking for
+        cy.wait('@multiPageQuery')
+          .its('requestBody.count')
+          .should('eq', 5);
+      });
+
+      it('only lists twelve pages of results', () => {
+        cy.wait('@multiPageQuery');
+        cy.wait('@multiPageQuery');
+        cy.get('.bx--pagination__text').should('contain', '12');
+      });
+    });
+
+    describe('and items per page is set to 10', () => {
```

Comment:
Isn't this already covered by the previous `describe` and by checking the number of pages with the default page size?

Reply:
Not sure if this comment is still relevant after the last couple commits I put in. I didn't want to manually test all of the options in the Items per page dropdown, so I just picked 5 (the default size), 10, and 50, to get a good spread.